### ISSUE TITLE
Close release branches when official build is failing

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#62099]
I think we need to start aggressively shutting down release branch merges when the signed queue is down. e.g.
1. when a change in the product makes CI on GitHub pass but makes the official build fail, as well as
2. when an underlying queue issue makes the official build fail.

Particularly in scenario (1) it makes it difficult to get RPS data on the changes which have come in since the queue recovered.

I'm wondering if some kind of a "circuit breaker" GitHub check could help here:
- Check reports a passing status while the signed queue is healthy.
- Check changes to red if the most recent finished official build is a failure.
- We can hit a button if we want to manually flip the check back over to green, i.e. if we think the underlying issue is resolved.

**Comment by jaredpar** [comment]
> Check changes to red if the most recent finished official build is a failure.

What is our official build pass rate? I worry that if we make it red on a single failure it will cause a lot of pain in getting PRs merged. Wondering if something like consecutive failures will be a better metric.

**Comment by RikkiGibson** [comment]
Pass rate for main-vs-deps over the past month:

<img width="1299" alt="image" src="https://user-images.githubusercontent.com/5833655/175368798-2ca57e53-d85e-4b21-b7eb-2ff625f63254.png">

My read of this is that on a release branch, either we are passing 100%, or something has taken the whole queue down, dropping the pass rate to zero or something low depending on how much of the day the queue was down for.

**Comment by jaredpar** [comment]
That is more consistent than I would expect. I 'd be fine with starting on a single failure based on that.

**Comment by vatsalyaagrawal** [comment]
Agree. Locking down early will help us avoid high-cost investigations/lockdowns later in the cycle. 

Automating this scenario will be awesome.
